app/models/foreman_virt_who_configure/output_generator.rb

```diff
@@ -42,19 +79,29 @@ def virt_who_output
 rhsm_encrypted_password=$user_password
 rhsm_prefix=/rhsm
 EOF
+  if [ $? -ne 0 ]; then result_code=$(($result_code|#{error_code(:virt_who_config_file_issue)})); fi
 
-echo "Creating sysconfig virt-who configuration.."
-cat > /etc/sysconfig/virt-who << EOF
+  echo "Creating sysconfig virt-who configuration.."
+  cat > /etc/sysconfig/virt-who << EOF
 VIRTWHO_SATELLITE6=1
 VIRTWHO_DEBUG=#{config.debug? ? 1 : 0}
 VIRTWHO_INTERVAL=#{config.interval * 60}#{proxy_strings}
 EOF
+  if [ $? -ne 0 ]; then result_code=$(($result_code|#{error_code(:virt_who_sysconfig_file_issue)})); fi
+
+  echo "Enabling and restarting the virt-who service"
+  chkconfig virt-who on || $(($result_code|#{error_code(:virt_who_chkconfig_issue)}))
+  service virt-who restart || $(($result_code|#{error_code(:virt_who_service_issue)}))
+else
+  result_code=$(($result_code|#{error_code(:virt_who_too_old)}))
+fi
 
-echo "Enabling and restarting the virt-who service"
-chkconfig virt-who on
-service virt-who restart
+if [ $result_code -ne 0 ]; then
+  echo "Errors during configuration, return code is $result_code"
```

Comment:
I understand how the error codes work in the script. But can't we have both returned error code and meaningful message printed from the script? Hammer can solve the issue by reading the messages from api, but if users copy-paste and execute only the script then the information they get is very limited.

BTW see my first line of the previous comment. I'm afraid the return code of the script if you copy-paste it into a file is currently always 0.